Source/WebCore/xml/parser/XMLDocumentParser.h

```diff
@@ -153,7 +153,7 @@ class XMLDocumentParser final : public ScriptableDocumentParser, public PendingS
 
     xmlParserCtxtPtr context() const { return m_context ? m_context->context() : nullptr; };
 
-    LocalFrameView* m_view { nullptr };
+    RefPtr<LocalFrameView> m_view;
```

Comment:
Slightly worried about a reference cycle here given that the XMLDocumentParser is not a stack-only object and is held strongly by the Document.

Reply:
Document holds WeakPtr to Frame, and Frame holds RefPtr to LocalFrameView. So I think RefPtr is probably fine. But WeakPtr is safer against cycles, and more consistent with Document holding WeakPtr to Frame. So let's make m_view WeakPtr here.

Or you could go even further and have a "bool m_isInFrameView" data member instead of a pointer. As far as I can tell, XMLDocumentParser literally only uses m_view to remember if it was initialized with a FrameView.

Reply:
I like the "bool m_isInFrameView" idea, it is what I implemented, but if you prefer WeakPtr, that would work too.